src/generators/schema-gen.ts

```diff
@@ -0,0 +1,117 @@
+
+import {JsonSchema} from "../models/jsonSchema";
+
+const AdditionalProperties = "additionalProperties";
+const RequiredProperties   = "required";
+
+type Properties = {[property: string]: JsonSchema}
+
+const distinct = (array: Array<any>, discriminator: (item: any) => any): Array<any> => {
+    const known = {};
+    return array.filter(item => {
+        const discriminatorValue = discriminator(item);
```

Comment:
discriminatorValue must be a string, in order to do the check in line 13
the discriminator can then be defined to be (item: any) => string
or am I mistaken?